glib/src/variant_iter.rs

```diff
@@ -59,6 +60,73 @@ impl DoubleEndedIterator for VariantIter {
 
 impl ExactSizeIterator for VariantIter {}
 
+/// Iterator over items in a variant of type `as`.
+#[derive(Debug)]
+pub struct VariantStrIter<'a> {
```

Comment:
And in that case this should probably become a macro as the code will be the same, just the getter for each child value, the returned type and the name would be different for each iterator.

Could also be done generically but as long as it's only for 2-3 iterators...

Reply:
Yeah, I actually have another mostly-done PR to add `iter_get()` which would heavily overlap with this, but I realized I mostly needed it for `as` and then realized I wanted this special case to avoid cloning to `String`.

In the end...I'm not sure because what we may really want is the equivalent of `g_variant_get_fixed_array()` for e.g. `au` (a generalization of `.bytes()`) - but those are also pretty obscure IME.

My general rule of thumb is that duplicating (similar but not identical) code twice can be OK, but if it becomes 3 versions (and definitely more) then that's usually the line where it makes sense to factor out common bits.  Which in this case would have to be a macro because we can't do [generic borrowing iterators yet](https://blog.rust-lang.org/2021/08/03/GATs-stabilization-push.html).